src/objc/ns_object.rs

```diff
@@ -86,4 +87,10 @@ impl<'data> NSObject<'data> {
     pub fn mutable_copy(&self) -> Arc<Self> {
         unsafe { _msg_send_any_cached![self, mutableCopy] }
     }
+
+    /// Returns a string that describes the contents of this object.
+    #[inline]
+    pub fn description(&self) -> Arc<NSString> {
```

Comment:
This should be behind the `foundation` feature flag, to not require `foundation` when only using `objc`.

I also try to link to Apple's original documentation when I can.

```suggestion
    ///
    /// See [documentation](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418746-description?language=objc).
    #[cfg(feature = "foundation")]
    #[inline]
    pub fn description(&self) -> Arc<crate::foundation::NSString> {
```

Reply:
alright, i made those changes. i mistakenly clobbered the original commit, but added new ones for the suggested changes. do you want me to build a commit history from the original PR commit, keep what's in the PR now, or should i squash before your merge?